python/pyspark/sql/pandas/conversion.py

```diff
@@ -369,7 +373,10 @@ def _convert_from_pandas(self, pdf, schema, timezone):
                             pdf[field.name] = s
             else:
                 for column, series in pdf.iteritems():
-                    s = _check_series_convert_timestamps_tz_local(series, timezone)
+                    s = series
+                    should_localize = not self._is_timestamp_ntz_preferred()
+                    if should_localize and is_datetime64tz_dtype(s.dtype) and s.dt.tz is not None:
```

Comment:
Seems we already handle for `is_datetime64tz_dtype` in `_check_series_convert_timestamps_tz_local`? For non `is_datetime64tz_dtype` case we skip localization? Is it different to previous?

Reply:
Yes, it is being handled there: https://github.com/apache/spark/blob/0494dc90af48ce7da0625485a4dc6917a244d580/python/pyspark/sql/pandas/types.py#L284-L292

For non `is_datetime64tz_dtype`, we skip localization because timezone is unavailable when pandas comtains datetime with `object` type as an example: 

```python
>>> import pandas as pd
>>> import datetime
>>> s = pd.Series([datetime.datetime.now()])
>>> s.astype("object").dt
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/.../pandas/core/generic.py", line 5487, in __getattr__
    return object.__getattribute__(self, name)
  File "/.../pandas/core/accessor.py", line 181, in __get__
    accessor_obj = self._accessor(obj)
  File "/.../pandas/core/indexes/accessors.py", line 506, in __new__
    raise AttributeError("Can only use .dt accessor with datetimelike values")
AttributeError: Can only use .dt accessor with datetimelike values
```